common/djangoapps/mailchimp_pipeline/tasks.py

```diff
@@ -34,51 +35,81 @@ def update_org_details_at_mailchimp(org_label, org_type, work_area, list_id):
 
 @task()
 def update_enrollments_completions_at_mailchimp(list_id):
+
     """Task to send user enrollments & course completions details to MailChimp"""
     log.info("starting enrollments & completions sync")
-    users = User.objects.all()
-
-    for user in users:
-        profile = user.profile
-        extended_profile = user.extended_profile
 
-        focus_areas = FocusArea.get_map()
-        org_sectors = OrgSector.get_map()
+    cursor = connection.cursor()
+    cursor.execute('SET TRANSACTION ISOLATION LEVEL READ COMMITTED')
 
-        org_type = ""
-        if extended_profile.organization and extended_profile.organization.org_type:
-            org_type = org_sectors.get(extended_profile.organization.org_type, '')
+    batch_size = 100
+    total_user_count = User.objects.all().count()
+    page_count = total_user_count / batch_size
+    counter = 0
 
-        all_certs = []
+    while counter is not page_count + 1:
         try:
-            all_certs = certificate_api.get_certificates_for_user(user.username)
+            page_start = counter * batch_size
+            page_end = page_start + batch_size
+
+            users = list(User.objects.all()[page_start:page_end])
+            log.info("Syncing batch from {} to {}".format(page_start, page_end))
+
+            for user in users:
+                profile = user.profile
+                extended_profile = user.extended_profile
+
+                focus_areas = FocusArea.get_map()
+                org_sectors = OrgSector.get_map()
```

Comment:
focus_areas and org_sectors should be out of loop as these are not related to user and contains global values